application_sdk/clients/temporal.py

```diff
@@ -271,27 +289,144 @@ def get_namespace(self) -> str:
         """
         return self.namespace
 
+    def _calculate_refresh_interval(self) -> int:
+        """Calculate the optimal token refresh interval based on token expiry.
+
+        Returns:
+            int: Refresh interval in seconds
+        """
+        # Try to get token expiry time
+        expiry_time = self.auth_manager.get_token_expiry_time()
+        if expiry_time:
+            # Calculate time until expiry
+            time_until_expiry = self.auth_manager.get_time_until_expiry()
+            if time_until_expiry and time_until_expiry > 0:
+                # Refresh at 80% of the token lifetime, but at least every 5 minutes
+                # and at most every 30 minutes
+                refresh_interval = max(
+                    5 * 60,  # Minimum 5 minutes
+                    min(
+                        30 * 60,  # Maximum 30 minutes
+                        int(time_until_expiry * 0.8),  # 80% of token lifetime
+                    ),
+                )
+                return refresh_interval
+
+        # Default fallback: refresh every 14 minutes
+        logger.info("Using default token refresh interval: 14 minutes")
+        return 14 * 60
+
+    async def _token_refresh_loop(self) -> None:
```

Comment:
Need a deeper dive on this

Reply:
Is there a way we can intercept each request being made by the client
- This can be subsclassing TemporalClient and creating our own

Reply:
class AtlanTemoralClient(TemporalClient):
   def make_request():
      .....
      super(rpc_metadata=...)